networks/rpc/client.go

```diff
@@ -733,6 +903,42 @@ func (c *Client) read(conn net.Conn) error {
 	}
 }
 
+// readGorillaWS is called on a dedicated goroutine.
+
+func (c *Client) readGorillaWS(conn *gorillaws.Conn) error {
+
+	readMessage := func() (rs []*jsonrpcMessage, err error) {
+		_, reader, err := conn.NextReader()
+		if err != nil {
+			return nil, err
+		}
+		var (
+			buf json.RawMessage
+			dec = json.NewDecoder(reader)
+		)
```

Comment:
Can I ask why you move those line to here?

Reply:
Variable dec needs a reader. The reader needs the result of conn.NextReader. Thus, var dec must be inside of readMessage() method.

Reply:
Variable dec needs a reader. The reader needs the result of conn.NextReader. Thus, var dec must be inside of readMessage() method.